### PR TITLE
Memoize Rosetta API GraphQL requests for genesis info

### PR DESCRIPTION
**Pull request by joaosreis**
Due to the increased size of the genesis ledger post-hard fork, GraphQL queries for genesis info are significantly taking more time to be responded to. This affects Rosetta API requests that rely on these queries, to the point that concurrent Rosetta requests make the GraphQL endpoint unresponsive.

To mitigate this, this PR implements the preexisting memoization mechanism in all Rosetta API GraphQL queries that retrieve genesis info. After caching the GraphQL requests, subsequent Rosetta requests are responded to in a similar time as compared to when the genesis ledger is smaller.

This has been tested by running an updated Rosetta node connected to the post-hard fork test network and validating it with `rosetta-cli`.

### CODE REVIEW

**Comment by joaosreis** [comment]
!ci-build-me

**Comment by joaosreis** [comment]
!ci-nightly-me

File: src/app/rosetta/lib/network.ml [hunk 1]
```diff
@@ -258,28 +289,32 @@ module Status = struct
...
+                  blockchainState = {utcDate = Int64.of_int_exn 1_594_854_566};
+                  consensusState = {blockHeight = Int64.of_int_exn 4 }
+                }
+                }|];
+          daemonStatus = {
+            peers = [|{peerId = "dev.o1test.net"}|]
+          };
+          syncStatus = `SYNCED;
+          initialPeers = [||];
+          networkID = "mina:testnet"
```
Comment (Sventimir) [inline src/app/rosetta/lib/network.ml]: Shouldn't this be computed dynamically like in other places?

**Review by Sventimir — APPROVED** [review]
Just one concern. Other than that it looks fine.

File: src/app/rosetta/lib/network.ml (hunk 1 above)
Comment (joaosreis) [reply src/app/rosetta/lib/network.ml]: I didn't change this logic, just had to indent the code. But this is to build the test setup, using this mock values.

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**